src/modules/dashboard/components/CreateEditDomainDialog/CreateEditDomainDialogForm.tsx

```diff
@@ -0,0 +1,161 @@
+import { canArchitect } from '../../../users/checks';
+
+import { FormValues } from './CreateEditDomainDialog';
+import styles from './CreateEditDomainDialogForm.css';
+
+const MSG = defineMessages({
+  titleCreate: {
+    id:
+      'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.titleCreate',
+    defaultMessage: 'Create a new domain',
+  },
+  titleEdit: {
+    id: 'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.titleEdit',
+    defaultMessage: 'Edit domain details',
+  },
+  name: {
+    id: 'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.name',
+    defaultMessage: 'Domain name',
+  },
+  purpose: {
+    id: 'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.name',
+    defaultMessage: 'What is the purpose of this domain?',
+  },
+  annotation: {
+    id:
+      'dashboard.CreateEditDomainDialog.CreateEditDomainDialogForm.annotation',
+    defaultMessage: 'Explain why you’re creating this domain',
+  },
+  noPermissionFrom: {
+    id:
+      'dashboard.CreatePaymentDialog.CreatePaymentDialogForm.noPermissionFrom',
+    defaultMessage:
+      // eslint-disable-next-line max-len
+      'You need the {roleRequired} permission in {domain} to take this action.',
+  },
+});
+
+interface Props {
+  back: () => void;
+  colony: Colony;
+  id?: string;
+}
+
+const CreateEditDomainDialogForm = ({
+  back,
+  colony,
+  handleSubmit,
+  id,
+}: Props & FormikProps<FormValues>) => {
+  const [domainColor, setDomainColor] = useState(Color.LightPink);
+
+  const { walletAddress, username, ethereal } = useLoggedInUser();
+
+  const allUserRoles = useTransformer(getAllUserRoles, [colony, walletAddress]);
+
+  const hasRegisteredProfile = !!username && !ethereal;
+  const canCreateEditDomain =
+    hasRegisteredProfile && canArchitect(allUserRoles);
+
+  return (
+    <>
+      <DialogSection>
+        <Heading
+          appearance={{ size: 'medium', margin: 'none' }}
+          text={id === undefined ? MSG.titleCreate : MSG.titleEdit}
+        />
+      </DialogSection>
+      {!canCreateEditDomain && (
+        <DialogSection>
+          <PermissionRequiredInfo requiredRoles={[ColonyRole.Architecture]} />
+        </DialogSection>
+      )}
+      <DialogSection>
+        <div className={styles.displayFlex}>
+          <div className={styles.domainName}>
+            <Input
+              label={MSG.name}
+              name="name"
+              appearance={{ colorSchema: 'grey', theme: 'fat' }}
+              disabled={!canCreateEditDomain}
+            />
+          </div>
+          <ColorSelect
+            activeOption={domainColor}
+            alignOptions="right"
+            onColorChange={setDomainColor}
+            disabled={!canCreateEditDomain}
+          />
+        </div>
+      </DialogSection>
+      <DialogSection>
+        <Input
+          label={MSG.purpose}
+          name="purpose"
+          appearance={{ colorSchema: 'grey', theme: 'fat' }}
+          maxLength={90}
```

Comment:
Why is there a max length enforcement here?  There is no cap on domain description lengths.

If this was spec'd in the designs that means the designs are wrong and need to be challenged.

Either remove this, or discuss with product why are we hard capping this _(especially since this can be bypassed creating domains directly on chain)_